app/models/agent.rb

```diff
@@ -55,6 +55,9 @@ class Agent < ApplicationRecord
   scope :with_territorial_role, lambda { |territory|
     joins(:territories).where(territories: { id: territory.id })
   }
+  scope :available_referents_for, lambda { |user|
+    where.not(id: [user.agents.map(&:id)])
```

Comment:
```suggestion
    where.not(id: user.agents.ids)
```
c’est un poil moins cher que `map(&:id)` 😄 . Il y avait aussi des `[]` en trop ou c’est moi qui n’ai pas bien compris?